### ISSUE TITLE
directory & filename hashing

### ISSUE THREAD

**Issue by c33s** [KnpLabs/Gaufrette#151]
to get better performance while serving a lot of files it would be better to split up files in multiple directories, like its done for php sessions:

> '5;/tmp'
> '/tmp/4/b/1/e/3/sess_4b1e384ad74619bd212e236e52a5a174If'

http://de2.php.net/manual/en/session.configuration.php#ini.session.save-path

> Keep in mind that on Linux if you have a directory with too many files, the shell may not be able to expand wildcards.

http://stackoverflow.com/a/466728

maybe also add a filename/key hashing option.

**Comment by wysow** [comment]
@c33s This could be closed right?

**Comment by c33s** [comment]
can it? isn't it only for the local adapter? shouldn't it be implemented for all/more?

**Comment by wysow** [comment]
@c33s Yes you're right, we can implement this in other adapters, feel free to send PRs if you can, meanwhile if you think what have already been done is sufficient for your current needs you can close this issue...
